File: api/schemas/tv.py
from pydantic import BaseModel


class TvHealth(BaseModel):
    state: str
    is_alarm: bool
    is_warning: bool
    message: str


class TopPourer(BaseModel):
    operator: str
    rate_lph: float


class PackingBreakdownRow(BaseModel):
    resin: str
    lot_number: str
    units: int
    color: str
    skids: float


class WorkOrderRow(BaseModel):
    resin_type: str
    color: str
    pump_station: str
    current_units: int
    target_units: int
    progress_pct: float


class TvOverviewOut(BaseModel):
    active_shift: str
    health: TvHealth
    current_output_l: float
    current_run_rate_lph: float
    expected_now_l: float
    target_lph: float
    pace_derived: bool
    pace_station_count: int
    pace_gap_l: float
    projected_total_l: float
    shift_target_l: float
    build_pct: float
    scope_packed: int
    total_poured: int
    total_packed: int
    unpacked_wip: int
    packing_enabled: bool
    show_runs_card: bool
    top_pourers: list[TopPourer]
    packing_breakdown: list[PackingBreakdownRow]
    work_orders: list[WorkOrderRow]

File: frontend/src/api/tv.ts
import { api } from './client'

export interface TvHealth {
  state: string
  is_alarm: boolean
  is_warning: boolean
  message: string
}

export interface TopPourer {
  operator: string
  rate_lph: number
}

export interface PackingBreakdownRow {
  resin: string
  lot_number: string
  units: number
  color: string
  skids: number
}

export interface WorkOrderRow {
  resin_type: string
  color: string
  pump_station: string
  current_units: number
  target_units: number
  progress_pct: number
}

export interface TvOverview {
  active_shift: string
  health: TvHealth
  current_output_l: number
  current_run_rate_lph: number
  expected_now_l: number
  target_lph: number
  pace_derived: boolean
  pace_station_count: number
  pace_gap_l: number
  projected_total_l: number
  shift_target_l: number
  build_pct: number
  scope_packed: number
  total_poured: number
  total_packed: number
  unpacked_wip: number
  packing_enabled: boolean
  show_runs_card: boolean
  top_pourers: TopPourer[]
  packing_breakdown: PackingBreakdownRow[]
  work_orders: WorkOrderRow[]
}

export const tvApi = {
  overview: (mode: 'auto' | 'daily') => api.get<TvOverview>(`/tv/overview?mode=${mode}`),
}
